tests/basic_test.go

```diff
@@ -162,7 +199,7 @@ func testPolicyWithFilterCaptureAndDesiredStateWithoutRef(t *testing.T) {
 			MetaInfo: types.MetaInfo{
 				Version: "0",
 			},
-			DesiredState: mainCurrentState,
+			DesiredState: mainExpectedDesiredState,
 			Cache: types.CachedState{
```

Comment:
So it would be nice to have the same with a cache:) I believe the PR has a bug if the desired state has a reference to a cached capture and the tests missed it.

Reply:
Did you add similar test with a cache? Couldn't find it.

Reply:
We are still missing 2 important tests.
1. Reference from the desired state to a cached capture.
2. Reference from a new capture to a cached capture.

Reply:
I was doing the test with empty `capture` field and some cached captured states but this line https://github.com/nmstate/nmpolicy/blob/main/nmpolicy/internal/capture/capture.go#L61 is removing from the cache the entries that does not have a capture entry expression.